src/rpp/rpp/operators/first.hpp

```diff
@@ -0,0 +1,75 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2022 - present.
+//                            TC Wang 2022 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/operators/fwd/first.hpp>
+#include <rpp/operators/take.hpp>
+#include <rpp/subscribers/constraints.hpp>
+#include <rpp/utils/functors.hpp>
+#include <rpp/utils/utilities.hpp>
+
+#include <memory>
+
+IMPLEMENTATION_FILE(first_tag);
+
+namespace rpp::details
+{
+
+template<constraint::decayed_type Type>
+struct first_impl
+{
+private:
+    struct first_value_forwarder
+    {
+        void forward_on_next(auto&& value, const constraint::subscriber auto& subscriber) const
+        {
+           m_take_on_next(std::forward<decltype(value)>(value), subscriber);
+        }
+
+        void forward_on_completed(const constraint::subscriber auto& subscriber) const
+        {
+            if (!subscriber.is_subscribed()) return;
+
+            if (m_take_on_next.emission_count() != 0)
+            {
+                auto err = std::make_exception_ptr(std::runtime_error{"there should be one emission before completed"});
```

Comment:
Can you create separate type of exception for this one, place it inside "fwd/first.hpp" and add comment inside documentation for "first" that it can \throws such type of exception?